linkerd/stack-metrics/src/lib.rs

```diff
@@ -0,0 +1,215 @@
+#![deny(warnings, rust_2018_idioms)]
+
+use futures::{Async, Poll};
+use indexmap::IndexMap;
+use linkerd2_metrics::{metrics, Counter, FmtLabels, FmtMetrics};
+use std::fmt;
+use std::hash::Hash;
+use std::sync::{Arc, Mutex};
+use std::time::Instant;
+
+metrics! {
+    stack_create_total: Counter { "Total number of services created" },
+    stack_drop_total: Counter { "Total number of services dropped" },
+    stack_poll_total: Counter { "Total number of stack polls" },
+    stack_poll_total_ms: Counter { "Total number of milliseconds this service has spent awaiting readiness" }
+}
+
+type Registry<L> = Arc<Mutex<IndexMap<L, Arc<Metrics>>>>;
+
+#[derive(Debug)]
+pub struct NewLayer<L: Hash + Eq> {
```

Comment:
Similar to comment on #417, it feels like this can be similar to the types in that [change](https://github.com/linkerd/linkerd2-proxy/blob/d5cf282ab3297dba0e83e7c2054096c81b6fd46d/linkerd/error-metrics/src/lib.rs#L29-L37). `NewLayer` is a little surprising to see in `core` code when all the other metrics are not named this way.